libvast/vast/system/source.hpp

```diff
@@ -158,22 +158,23 @@ struct source_state {
     if (type_registry) {
       self->request(type_registry, caf::infinite, atom::get_v)
         .await([=](type_set types) {
-          auto& st = selfptr->state;
           auto is_valid = [&](const auto& layout) {
             return detail::starts_with(layout.name(), type_filter);
           };
           // First, merge and de-duplicate the local schema with types from the
           // type-registry.
-          types.insert(std::make_move_iterator(st.local_schema.begin()),
-                       std::make_move_iterator(st.local_schema.end()));
-          st.local_schema.clear();
+          schema s;
+          for (auto& type : local_schema)
+            if (auto&& layout = caf::get_if<vast::record_type>(&type))
+              if (is_valid(*layout))
+                s.add(std::move(*layout));
```

Comment:
Can you drop this commit? It is entirely unrelated from my PR and we said yesterday that we want to do this in a separate PR.

Not quite sure why you've just pushed this on top, because we were already done reviewing the logic and now there needs to be yet another reviewer.

Reply:
I added this because it is **actually required** to fix the reported issue. If we don't give preference to the local schema at the source there is no way it can ever make its way into the type registry without a restart of the node.